v2/blockstore/readonly.go

```diff
@@ -152,6 +170,9 @@ func (b *ReadOnly) PutMany([]blocks.Block) error {
 
 // AllKeysChan returns the list of keys in the CAR.
 func (b *ReadOnly) AllKeysChan(ctx context.Context) (<-chan cid.Cid, error) {
+	b.mu.RLock()
+	defer b.mu.RUnlock()
+
```

Comment:
Do we also need to take the read lock inside the go-routine?

Reply:
Yikes, good catch. We probably want to grab the mutex here, and release it when the other goroutine ends.

Reply:
Done.